.github/workflows/k8s_integration_test.yml

```diff
@@ -0,0 +1,128 @@
+    inputs:
+      spark-ref:
+        description: 'Specified PR Ref/SHA/Branch/Tag:'
+        required: true
+        default: 'master'
+      sync-current:
+        type: choice
+        description: 'Sync the current branch:'
+        required: true
+        options:
+          - 'true'
+          - 'false'
+
+jobs:
+  k8s-integration-tests:
+    name: "Run Spark on Kubernetes Integration test (${{ github.event.inputs.spark-ref || github.event.ref }})"
+    runs-on: ubuntu-20.04
+    steps:
+      # Checkout based branch
+      - name: Checkout Spark repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          repository: apache/spark
+          # Priority:
+          # 1. workflow_dispatch input
+          # 2. push.ref in apache projects for push event post validation
+          # 3. master, current branch for schedule or when we enable trigger for each pr
+          ref: >-
+            ${{
+              github.event.inputs.spark-ref
+              || (github.repository == 'apache/spark' && github.event.ref)
+              || 'master'
+            }}
+      - name: Sync the current branch with the latest in Apache Spark
+        # Do sync when workflow_dispatch(with sync-current=true) or other left events like push/schedule
+        if: ${{ github.event_name != 'workflow_dispatch' || github.event.inputs.sync-current == 'true' }}
+        run: |
+          echo "APACHE_SPARK_REF=$(git rev-parse HEAD)" >> $GITHUB_ENV
+          git fetch https://github.com/$GITHUB_REPOSITORY.git ${GITHUB_REF#refs/heads/}
+          git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' merge --no-commit --progress --squash FETCH_HEAD
+          git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' commit -m "Merged commit"
+      - name: Cache Scala, SBT and Maven
+        uses: actions/cache@v2
+        with:
+          path: |
+            build/apache-maven-*
+            build/scala-*
+            build/*.jar
+            ~/.sbt
+          key: build-${{ hashFiles('**/pom.xml', 'project/build.properties', 'build/mvn', 'build/sbt', 'build/sbt-launch-lib.bash', 'build/spark-build-info') }}
+          restore-keys: |
+            build-
+      - name: Cache Coursier local repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/coursier
+          key: k8s-integration-coursier-${{ hashFiles('**/pom.xml', '**/plugins.sbt') }}
+          restore-keys: |
+            k8s-integration-coursier-
+      - name: Install Java 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: start minikube
+        run: |
+          # See more in "Installation" https://minikube.sigs.k8s.io/docs/start/
+          curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+          sudo install minikube-linux-amd64 /usr/local/bin/minikube
+          minikube start --cpus 2 --memory 6144
```

Comment:
Sorry, I forgot the context. Is this the maximum we can use?

Reply:
yep, 6947 is maximum according my test, but this is not integer multiple of 1024, so I just set this to 6144 for better resource statistic.

```
X Exiting due to RSRC_OVER_ALLOC_MEM: Requested memory allocation 7168MB is more than your system limit 6947MB.
```

[1] https://github.com/Yikun/spark/runs/5625298594?check_suite_focus=true#step:7:22

Reply:
Could you add a comment explicitly about that, @Yikun  ?

Reply:
added: `# system limit cpu:2, memory: 6947MB`